### ISSUE TITLE
Dashboard default date value in query missing

### ISSUE THREAD

**Issue by KEN-LJQ** [metabase/metabase#12362]
**Describe the bug**
In dashboard we have a date filter reference to a column of type `unixtime(second) `, can set the default value like `past 1 days`. But when we save and open the dashboard from homepage, the default value not working, also we notice that original request was canceled, and then resend a new one without default value.
![image](https://user-images.githubusercontent.com/16650282/79817766-c6c14f00-83b8-11ea-9e52-906b570bbce8.png)

**Logs**
![image](https://user-images.githubusercontent.com/16650282/79818058-70a0db80-83b9-11ea-9e79-90fd12eda2db.png)

```
app-main.bundle.js?1636e40125c6bad14763:5 Error getting setting scalar.field TypeError: Cannot read property 'cols' of undefined
    at Object.getDefault (app-main.bundle.js?1636e40125c6bad14763:5)
    at g (app-main.bundle.js?1636e40125c6bad14763:5)
    at t.getComputedSettings (app-main.bundle.js?1636e40125c6bad14763:5)
    at h (app-main.bundle.js?1636e40125c6bad14763:5)
    at t.value (app-main.bundle.js?1636e40125c6bad14763:5)
    at t.value (app-main.bundle.js?1636e40125c6bad14763:5)
    at c.performInitialMount (vendor.bundle.js?1636e40125c6bad14763:1)
    at c.mountComponent (vendor.bundle.js?1636e40125c6bad14763:1)
    at Object.mountComponent (vendor.bundle.js?1636e40125c6bad14763:1)
    at c.performInitialMount (vendor.bundle.js?1636e40125c6bad14763:1)
g @ app-main.bundle.js?1636e40125c6bad14763:5
t.getComputedSettings @ app-main.bundle.js?1636e40125c6bad14763:5
h @ app-main.bundle.js?1636e40125c6bad14763:5
value @ app-main.bundle.js?1636e40125c6bad14763:5
value @ app-main.bundle.js?1636e40125c6bad14763:5
performInitialMount @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
performInitialMount @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_createInitialChildren @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
performInitialMount @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_createInitialChildren @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_createInitialChildren @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_createInitialChildren @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
performInitialMount @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
performInitialMount @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_createInitialChildren @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
mountComponent @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_reconcilerUpdateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateDOMChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_reconcilerUpdateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateDOMChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_reconcilerUpdateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateChildren @ vendor.bundle.js?1636e40125c6bad14763:1
_updateDOMChildren @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
receiveComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_updateRenderedComponent @ vendor.bundle.js?1636e40125c6bad14763:1
_performComponentUpdate @ vendor.bundle.js?1636e40125c6bad14763:1
updateComponent @ vendor.bundle.js?1636e40125c6bad14763:1
performUpdateIfNecessary @ vendor.bundle.js?1636e40125c6bad14763:1
performUpdateIfNecessary @ vendor.bundle.js?1636e40125c6bad14763:1
b @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
M @ vendor.bundle.js?1636e40125c6bad14763:1
closeAll @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
batchedUpdates @ vendor.bundle.js?1636e40125c6bad14763:1
e @ vendor.bundle.js?1636e40125c6bad14763:1
a @ vendor.bundle.js?1636e40125c6bad14763:1
enqueueSetState @ vendor.bundle.js?1636e40125c6bad14763:1
s.setState @ vendor.bundle.js?1636e40125c6bad14763:1
i.onStateChange @ vendor.bundle.js?1636e40125c6bad14763:1
notify @ vendor.bundle.js?1636e40125c6bad14763:1
e.notifyNestedSubs @ vendor.bundle.js?1636e40125c6bad14763:1
i.onStateChange @ vendor.bundle.js?1636e40125c6bad14763:1
g @ vendor.bundle.js?1636e40125c6bad14763:1
(anonymous) @ vendor.bundle.js?1636e40125c6bad14763:1
(anonymous) @ vendor.bundle.js?1636e40125c6bad14763:1
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
dispatch @ vendor.bundle.js?1636e40125c6bad14763:1
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
u @ vendor.bundle.js?1636e40125c6bad14763:1
g._invoke @ vendor.bundle.js?1636e40125c6bad14763:1
e.<computed> @ vendor.bundle.js?1636e40125c6bad14763:1
r @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
Promise.then (async)
r @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
a.type @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ vendor.bundle.js?1636e40125c6bad14763:1
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
u @ vendor.bundle.js?1636e40125c6bad14763:1
g._invoke @ vendor.bundle.js?1636e40125c6bad14763:1
e.<computed> @ vendor.bundle.js?1636e40125c6bad14763:1
r @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
t @ app-main.bundle.js?1636e40125c6bad14763:5
(anonymous) @ app-main.bundle.js?1636e40125c6bad14763:5
value @ app-main.bundle.js?1636e40125c6bad14763:5
e.notifyAll @ vendor.bundle.js?1636e40125c6bad14763:1
close @ vendor.bundle.js?1636e40125c6bad14763:1
closeAll @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
M @ vendor.bundle.js?1636e40125c6bad14763:1
closeAll @ vendor.bundle.js?1636e40125c6bad14763:1
perform @ vendor.bundle.js?1636e40125c6bad14763:1
batchedUpdates @ vendor.bundle.js?1636e40125c6bad14763:1
batchedUpdates @ vendor.bundle.js?1636e40125c6bad14763:1
dispatchEvent @ vendor.bundle.js?1636e40125c6bad14763:1
```
The canceled request:
![image](https://user-images.githubusercontent.com/16650282/79816776-95e01a80-83b6-11ea-92ac-ee60ae8314ec.png)
The resend request which missing default value
![image](https://user-images.githubusercontent.com/16650282/79816816-b1e3bc00-83b6-11ea-9cdd-e631e44908c1.png)

**To Reproduce**
1. Create a dashboard
2. Add date filter reference to a data with type `unixtime(second)`
3. Set the default value of date fileter
4. Go to homepage and enter this dashboard
5. See error

**Expected behavior**
Query and show the data with default filter value when first enter the dashboard.

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "zh-CN",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_162-b12",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_162",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "25.162-b12",
    "os.name": "Linux",
    "os.version": "2.6.32-696.3.1.el6.x86_64",
    "user.language": "zh",
    "user.timezone": "Asia/Shanghai"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "presto"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.26-log"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.35.2",
      "date": "2020-04-10",
      "branch": "release-0.35.x",
      "hash": "f3b2857"
    },
    "settings": {
      "report-timezone": "Asia/Hong_Kong"
    }
  }
}
```

**Severity**
Without the default value, when open dashboard of some big table, it will query all data. And user have an bad experience because have to select the filter again

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Hi @KEN-LJQ 
I cannot reproduce your issue.
1. Are you querying Presto?
2. Are you able to reproduce this with Sample Dataset? That would help a lot, since we have all access to that.
3. The `(canceled)` requests are canceled by your browser - not by Metabase.
Could it be an extension in Chrome that is doing something it shouldn't? Or perhaps automatic translation?
4. Is the `openTime` filter empty/blank, like you're showing in the screenshot or did you redact it?
5. What does the URL look like, when you access the dashboard?
Should be `http://example.com/dashboard/245?openTime=past1days`

**Comment by KEN-LJQ** [comment]
@flamber 
1. Yes, we use presto
2. We can reproduce with Sample DataSet
3. We have tested is within private mode withou any extention, issue still happend
4. The `openTime` filter had set default value, but when you open dashboard from collections, it is empty like you ses in the screenshot
5. Yes, when access dashboard the url is `http://example.com/dashboard/245?openTime=past1days`, but soon, the parameter of url has gone and fiter become empty, at the same time, we can see some `canceled` reqeust

Here is how  we reproduce with Sample DataSet:
The key point is: we use unixtime, interger type to represent datetime, like `1587564326`
1. Use `Order` table, set data type of `quantity` to `UNIX Timestamp(Senonds)`
![image](https://user-images.githubusercontent.com/16650282/79992266-d5595480-84e5-11ea-8053-4012b3975be4.png)
2. Create a question, just simple count line number
![image](https://user-images.githubusercontent.com/16650282/79992413-09347a00-84e6-11ea-845a-c04e92f187f3.png)
3. Create a dashboard with this question, add a filter and set the default value of `Previous1Day`, then save
![image](https://user-images.githubusercontent.com/16650282/79992881-a8f20800-84e6-11ea-97f8-4d078b353e50.png)
4. exit, and open dashboard from collections, then you can see that
![image](https://user-images.githubusercontent.com/16650282/79992791-8bbd3980-84e6-11ea-8e3f-c4e812930afb.png)

we hope you can reproduce with these steps, wating for your news

**Comment by flamber** [comment]
@KEN-LJQ Perfect :+1: I was able to reproduce it with your new instructions (specially step 1)

**Comment by KEN-LJQ** [comment]
Do you have any workaround?

**Comment by flamber** [comment]
@KEN-LJQ Other than creating a database view, where you change the column to a regular timestamp or datetime, then no.

**Comment by KEN-LJQ** [comment]
Thanks, will this issue be fix in next version?

**Comment by flamber** [comment]
@KEN-LJQ No, most likely not, there's 2000+ other issues open, and many with much higher priority.

**Comment by Imaginons** [comment]
I'm facing the same issue. Another symptom of this issue is that the dashboard can't accept a date parameter. I suspect the dashboard filter is waiting for an Integer (unixtime) instead of a date. Any date paramater will be trunctated to the first digits before the dash (2020-04-26 will be truncated to 2020) causing the dashboard to display to its default.

A proposed workaround is to create a database view using the FROM_UNIXTIME() function from MySQL (or similar function in other databases) to transform the Unixtime format into a valid date format. Then change all your dashboard questions to use your new view date field instead of the original table field. The dashboard date filter can now use a default value and the URL will also accept a date parameter.

**Comment by flamber** [comment]
Upping priority, since it breaks drill-through by not passing the filter

**Comment by paulrosenzweig** [comment]
Closed by #12983
